src/math/hough_transform_simd.cpp

```diff
@@ -0,0 +1,271 @@
+#include "hough_transform_simd.h"
+#include "../image_function_helper.h"
+#include "../penguinv/cpu_identification.h"
+
+#ifdef PENGUINV_AVX_SET
+#include <immintrin.h>
+#endif
+
+#ifdef PENGUINV_SSE_SET
+#include <emmintrin.h>
+#endif
+
+#ifdef PENGUINV_NEON_SET
+#include <arm_neon.h>
+#endif
+
+#include <algorithm>
+#include <cmath>
+
+namespace
+{
+    const bool float_layout_check  = std::is_standard_layout<PointBase2D<float>>::value;
+    const bool double_layout_check = std::is_standard_layout<PointBase2D<double>>::value;
+    const uint32_t avx_double = 4u;
+    const uint32_t avx_float = 8u;
+    const uint32_t sse_double = 2u;
+    const uint32_t sse_float = 4u;
+    const uint32_t neon_double = 2u;
+    const uint32_t neon_float = 4u;
+
+    const float minimumAngleStep = 0.001f * static_cast<float>( pvmath::pi ) / 180.0f;
+    const float minimumLineTolerance = 1e-5f;
+    
+    template <typename _Type>
+    void FindDistance( const std::vector< PointBase2D< _Type > > & input, std::vector < _Type > & distance, _Type cosVal, _Type sinVal, const size_t inputPointCount )
+    {
+        _Type * distanceVal = distance.data();
+        const PointBase2D<_Type> * point = input.data();
+        const PointBase2D<_Type> * pointEnd = point + inputPointCount;
+
+        for ( ; point != pointEnd; ++point, ++distanceVal )
+            (*distanceVal) = point->x * sinVal + point->y * cosVal;
+    }
+
+    template <typename _Type>
+    void FindDistanceSimd( const std::vector< PointBase2D< _Type > > & input, std::vector < _Type > & distance, _Type cosVal, _Type sinVal, const size_t inputPointCount )
+    {
+        FindDistance( input, distance, cosVal, sinVal, inputPointCount );
+    }
+
+    template <>
+    void FindDistanceSimd< float >( const std::vector< PointBase2D< float > > & input, std::vector < float > & distance, float cosVal, float sinVal, const size_t inputPointCount )
+    {
+        
+        simd::SIMDType simdType = simd::actualSimdType();
+
+        if ( simdType == simd::avx_function && float_layout_check ) {
+            #ifdef PENGUINV_AVX_SET
+            const uint32_t simdWidth = (inputPointCount*2) / (avx_float*2);
+            const uint32_t totalSimdWidth = simdWidth * (avx_float*2);
```

Comment:
What is we set `avx_float`value in 2 times bigger initially? We know that we handle float[2] array always.

Reply:
no we also use `avx_float` in the for loop